integration/securitypolicy_test.go

```diff
@@ -804,3 +804,200 @@ func TestSecurityPolicyForGraphQL(t *testing.T) {
+				err := c.Client().Resources(testNs).Delete(ctx, policy)
+				eval.NoErr(err)
+
+				err = wait.For(
+					conditions.New(c.Client().Resources(testNs)).ResourceDeleted(policy),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				err = wait.For(func() (done bool, err error) {
+					oc := v1alpha1.OperatorContext{}
+
+					err = c.Client().Resources(testNs).Get(ctx, opCtx.Name, opCtx.Namespace, &oc)
+					if err != nil {
+						return false, err
+					}
+
+					if len(oc.Status.LinkedSecurityPolicies) > 0 {
+						return false, fmt.Errorf(
+							"failed to remove LinkedSecurityPolicy from OperatorContext, "+
+								"want: %v, got: %v, ", 0, len(oc.Status.LinkedSecurityPolicies),
+						)
+					}
+
+					return true, nil
+				}, wait.WithTimeout(defaultWaitTimeout), wait.WithInterval(defaultWaitInterval))
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("Recreate SecurityPolicy and ensure the link between resources",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				testNs, ok := ctx.Value(ctxNSKey).(string)
+				eval.True(ok)
+
+				var err error
+				policy, err = createTestPolicy(ctx, c, testNs, func(p *v1alpha1.SecurityPolicy) {
+					p.Spec.Context = &model.Target{Name: opCtx.Name, Namespace: opCtx.Namespace}
+				})
+				eval.NoErr(err)
+
+				err = wait.For(conditions.New(c.Client().Resources()).
+					ResourceMatch(policy, func(object k8s.Object) bool {
+						pol, ok := object.(*v1alpha1.SecurityPolicy)
+						if !ok {
+							return false
+						}
+
+						if pol.Spec.Context == nil || pol.Spec.Context.Name == "" || pol.Spec.Context.Namespace == "" {
+							return false
+						}
+
+						return pol.Spec.Context.Name == opCtx.Name && pol.Spec.Context.Namespace == opCtx.Namespace
+					}), wait.WithTimeout(defaultWaitTimeout), wait.WithInterval(defaultWaitInterval))
+				eval.NoErr(err)
+
+				err = wait.For(conditions.New(c.Client().Resources()).
+					ResourceMatch(opCtx, func(object k8s.Object) bool {
+						oc, ok := object.(*v1alpha1.OperatorContext)
+						if !ok {
+							return false
+						}
+
+						if len(oc.Status.LinkedSecurityPolicies) != 1 {
+							return false
+						}
+
+						linkedPolMeta := oc.Status.LinkedSecurityPolicies[0]
+
+						return linkedPolMeta.Name == policy.Name && linkedPolMeta.Namespace == policy.Namespace
+					}), wait.WithTimeout(defaultWaitTimeout), wait.WithInterval(defaultWaitInterval))
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("Break the link between resources by removing reference",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				testNs, ok := ctx.Value(ctxNSKey).(string)
+				eval.True(ok)
+
+				policy.Spec.Context = nil
+				err := c.Client().Resources(testNs).Update(ctx, policy)
+				eval.NoErr(err)
+
+				err = wait.For(conditions.New(c.Client().Resources()).
```

Comment:
Maybe we can skip this check. Since we are just setting field to nil and don't perform any operation in the controller.

Reply:
yes, you are right. i'll update the test case accordingly

Reply:
@komalsukhani, on second thought, maybe we can keep this test case here. i remember my goal here to understand that setting `contextRef` to nil won't change in later steps. I mean, we have a logic where we handle deleting/adding context references to resources. so, I just wanted to ensure that after deleting a `contextRef`, the controller won't add the same contextRef to resources in subsequent reconciliations. what do you think

Reply:
Makes sense